### PR TITLE
Couple of minor cleanups to prep for fabric PR

### PR DESCRIPTION
**Pull request by acoates-ms**
When bringing in more code for Fabric, there is a version of ShadowNode for fabric, so I'm renaming the file for paper to avoid collisions.

Moving the root tag generation to a separate class so that rootview tag generation can be shared between paper and fabric and avoid collusions between the two uimangers.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/6440)

### CODE REVIEW

File: vnext/Microsoft.ReactNative/Microsoft.ReactNative.vcxproj [hunk 1]
```diff
@@ -318,6 +319,7 @@
     <ClInclude Include="Views\Impl\SnapPointManagingContentControl.h" />
     <ClInclude Include="Views\IXamlRootView.h" />
     <ClInclude Include="Views\KeyboardEventHandler.h" />
+    <ClInclude Include="Views\PaperShadowNode.h" />
```
Comment (dannyvv) [inline vnext/Microsoft.ReactNative/Microsoft.ReactNative.vcxproj:322]: We should make sure that we also update the nuspec for the Win32 package....
The file is copied in [Layout-Headers.ps1#L69](https://github.com/microsoft/react-native-windows/blob/master/vnext/Scripts/Tfs/Layout-Headers.ps1#L69)
The file is then packaged in: [OfficeReact.Win32.nuspec#L51](https://github.com/microsoft/react-native-windows/blob/master/vnext/Scripts/OfficeReact.Win32.nuspec#L51)

Comment (dannyvv) [reply vnext/Microsoft.ReactNative/Microsoft.ReactNative.vcxproj:322]: Note: This might require changes in Office code as well...

**Review by dannyvv — APPROVED** [review]

File: vnext/Microsoft.ReactNative/Microsoft.ReactNative.vcxproj (hunk 1 above)
Comment (acoates-ms) [reply vnext/Microsoft.ReactNative/Microsoft.ReactNative.vcxproj:322]: Thats a different shadownode file I believe.  (I know.. too many of them around)

**Comment by asklar** [comment]
> When bringing in more code for Fabric, there is a version of ShadowNode for fabric, so I'm renaming the file for paper to avoid collisions.
> 
> Moving the root tag generation to a separate class so that rootview tag generation can be shared between paper and fabric and avoid collusions between the two uimangers.
> 
> ###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/6440)

is that "paper" the same as https://callstack.github.io/react-native-paper/ ?

File: vnext/Microsoft.ReactNative/Modules/NativeUIManager.cpp [hunk 2]
```diff
@@ -183,8 +184,7 @@ void NativeUIManager::destroyRootShadowNode(ShadowNode *node) {
 }
 
 int64_t NativeUIManager::AddMeasuredRootView(facebook::react::IReactRootView *rootView) {
-  auto tag = m_nextRootTag;
-  m_nextRootTag += RootViewTagIncrement;
+  auto tag = getNextRootViewTag();
```
Comment (asklar) [inline vnext/Microsoft.ReactNative/Modules/NativeUIManager.cpp:187]: const?

**Review by asklar — APPROVED** [review]

**Comment by acoates-ms** [comment]
> > When bringing in more code for Fabric, there is a version of ShadowNode for fabric, so I'm renaming the file for paper to avoid collisions.
> > Moving the root tag generation to a separate class so that rootview tag generation can be shared between paper and fabric and avoid collusions between the two uimangers.
> > ###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/6440)
> 
> is that "paper" the same as https://callstack.github.io/react-native-paper/ ?

No paper in this case is what the "old" non-fabric uimanager is called.
https://github.com/facebook/react-native/blob/master/Libraries/ReactNative/PaperUIManager.js

**Pull request merged**